src/Model/BlockInterface.php

```diff
@@ -35,7 +35,7 @@ public function getId();
     /**
      * Sets the name.
      */
-    public function setName(string $name): void;
+    public function setName(?string $name): void;
```

Comment:
This change breaks BC.

Reply:
Strictly speaking yes but is this not a bug that was created whilst adding types? I see no reason why this may not be set back to `null` (its original value).

Reply:
I didn't analyze the reason, but any class implementing this interface will start failing after this change. That's is what I'm pointing here. I think this is not a matter about the strictness level.

Reply:
@phansys I've reverted this change. It's also a bit unclear to me what the `name` of a block represents or what its intended purpose is. Perhaps someone can clarify? E.g. we've used it as a form field (like `content`) hence we're having trouble updating this to `null`. I can also move that to `settings` instead so the values can be nullified if required. Just by looking at the model itself tho it feels strange that it can be constructed without a name but cannot be reset to `null` via its setter.

Note that the `getTemplate` signature change is still there. That breaks BC too, no? So not sure what we're going to do with that. 

Hopefully we can merge the other fixes, if not, we can still work around it but thought I'd share this first.

Reply:
>Note that the getTemplate signature change is still there. That breaks BC too, no?

While we should consider the change at `getTemplate()` as a BC break, at least it will not produce a PHP error.